cmake/Modules/FindAzureSDK_EP.cmake

```diff
@@ -83,38 +85,86 @@ if (NOT AZURESDK_FOUND)
-      LOG_OUTPUT_ON_FAILURE ${TILEDB_LOG_OUTPUT_ON_FAILURE}
-      DEPENDS ${DEPENDS}
+    
+    if (WIN32)
+        # needed for applying patches on windows
+        find_package(Git REQUIRED)
+        #see comment on this answer - https://stackoverflow.com/a/45698220
+        #and this - https://stackoverflow.com/a/62967602 (same thread, different answer/comment)
+    else()
+        set(PATCH patch -N -p1)
+    endif()
+    
+    if(WIN32)
+      ExternalProject_Add(ep_azuresdk
+        PREFIX "externals"
+        URL "https://github.com/Azure/azure-storage-cpplite/archive/v0.2.0.zip"
+        URL_HASH SHA1=058975ccac9b60b522c9f7fd044a3d2aaec9f893
+        CMAKE_ARGS
+          -DVERBOSE=1
+          -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
+          -DBUILD_SHARED_LIBS=OFF
+          -DBUILD_TESTS=OFF
+          -DBUILD_SAMPLES=OFF
+          -DCMAKE_PREFIX_PATH=${TILEDB_EP_INSTALL_PREFIX}
+          -DCMAKE_INSTALL_PREFIX=${TILEDB_EP_INSTALL_PREFIX}
+          -DCMAKE_CXX_FLAGS=${CXXFLAGS_DEF}
+          -DCURL_INCLUDE_DIR=${TILEDB_EP_INSTALL_PREFIX}/include
+          -DCMAKE_C_FLAGS=${CFLAGS_DEF}
+        PATCH_COMMAND
+          echo starting patching for azure &&
+          cd ${CMAKE_SOURCE_DIR} &&
+          ${GIT_EXECUTABLE} apply --ignore-whitespace -p1 --unsafe-paths --verbose --directory=${TILEDB_EP_SOURCE_DIR}/ep_azuresdk < ${TILEDB_CMAKE_INPUTS_DIR}/patches/ep_azuresdk/remove-uuid-dep.patch &&
+          ${GIT_EXECUTABLE} apply --ignore-whitespace -p1 --unsafe-paths --verbose --directory=${TILEDB_EP_SOURCE_DIR}/ep_azuresdk < ${TILEDB_CMAKE_INPUTS_DIR}/patches/ep_azuresdk/azurite-support.patch &&
+          ${GIT_EXECUTABLE} apply --ignore-whitespace -p1 --unsafe-paths --check --apply --verbose --directory=${TILEDB_EP_SOURCE_DIR}/ep_azuresdk < ${TILEDB_CMAKE_INPUTS_DIR}/patches/ep_azuresdk/azure-storage-lite-base64.patch &&
+          ${GIT_EXECUTABLE} apply --ignore-whitespace -p1 --unsafe-paths --check --apply --verbose --directory=${TILEDB_EP_SOURCE_DIR}/ep_azuresdk < ${TILEDB_CMAKE_INPUTS_DIR}/patches/ep_azuresdk/azure-storage-lite-storage_url.patch &&
+          echo done patches for azure
+        LOG_DOWNLOAD TRUE
+        LOG_CONFIGURE TRUE
+        LOG_BUILD TRUE
+        LOG_INSTALL TRUE
+        LOG_OUTPUT_ON_FAILURE ${TILEDB_LOG_OUTPUT_ON_FAILURE}
+        DEPENDS ${DEPENDS}
     )
+    else()
+      ExternalProject_Add(ep_azuresdk
+        PREFIX "externals"
+        URL "https://github.com/Azure/azure-storage-cpplite/archive/v0.2.0.zip"
+        URL_HASH SHA1=058975ccac9b60b522c9f7fd044a3d2aaec9f893
+        CMAKE_ARGS
+          -DVERBOSE=1
+          -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
+          -DBUILD_SHARED_LIBS=OFF
+          -DBUILD_TESTS=OFF
+          -DBUILD_SAMPLES=OFF
+          -DCMAKE_PREFIX_PATH=${TILEDB_EP_INSTALL_PREFIX}
+          -DCMAKE_INSTALL_PREFIX=${TILEDB_EP_INSTALL_PREFIX}
+          -DCMAKE_CXX_FLAGS=${CXXFLAGS_DEF}
+          -DCURL_INCLUDE_DIR=${TILEDB_EP_INSTALL_PREFIX}/include
```

Comment:
Remove? (and above?) -- previously was building on posix without.

Reply:
(uncertain if you did not notice/highlight CURL_STATICLIB in CXXFLAGS_DEF also, but)
can remove CURL_STATICLIB from CFLAGS_DEF
(but is needed on windows on CXXFLAGS_DEF to link 
 see 5.7 of https://curl.se/docs/faq.html)

Removed -DCURL_INCLUDE_DIR=${TILEDB_EP_INSTALL_PREFIX}/include

Reply:
I'm not sure if this will work for conda, where we dynamically link libcurl. But I guess I can patch it if I have to. I don't think we dynamically link curl on windows anywhere else.

Reply:
We should be able to detect from our `FindCURL_EP` if it is a static library or a dynamic one right and conditionally set this?

Reply:
@ihnorton "conda, where we dynamically link libcurl"... are you referring to linking libcurl in tiledb? 
And, on conda, is libcurl already present in conda on windows, hence not fetched/built by superbuild[**]?
This location is specifically related to linking of azuresdk when superbuild fetches/builds azuresdk ...
And for superbuild with NOT CURL_FOUND, FindCurl_EP.cmake, prior to my changes, was already for WIN32 setting -DCURL_STATICLIB=ON, whatever that is accomplishing (building curl static lib?)...
----
[**](guess that may be (other) issue for azuresdk fetch/build, if libcurl dynamic is already present and therefore not fetched/built by tiledb superbuild...?)

Reply:
(Sorry for bold text, failing to find cause)
@ihnorton "conda, where we dynamically link libcurl",
are you referring to linking libcurl in tiledb? 
And, on conda, is libcurl already present in conda on windows, hence not fetched/built by superbuild[1]?
This location is specifically related to linking of azuresdk when superbuild fetches/builds azuresdk
And for superbuild with NOT CURL_FOUND, FindCurl_EP.cmake, prior to my changes, was already for WIN32 setting -DCURL_STATICLIB=ON, whatever that is accomplishing (building curl static lib?)
----
[1](guess that may be (other) issue for azuresdk fetch/build, if libcurl dynamic is already present and therefore not fetched/built by tiledb superbuild?)

Reply:
hmm, on further exploration, guess that's feeding thru to tiledb isn't it... as azuresdk apparently just builds a library.

Reply:
> are you referring to linking libcurl in tiledb?

yes

> And, on conda, is libcurl already present in conda on windows

yes

> guess that's feeding thru to tiledb isn't it

yes

The `CURL::libcurl` cmake target sets the `CURL_STATICLIB` definition on the cmake interface, so that it is used appropriately when needed:
https://github.com/TileDB-Inc/TileDB/blob/dev/cmake/Modules/FindCurl_EP.cmake#L207-L210

That interface is linked to the core tiledb objects here:

https://github.com/TileDB-Inc/TileDB/blob/dev/tiledb/CMakeLists.txt#L351-L360

But I believe the problem is that the azure SDK doesn't actually link `CURL::libcurl`:
https://github.com/Azure/azure-storage-cpplite/blob/master/CMakeLists.txt#L170-L171

So in that case, given that this is localized to the superbuild config, I think it's ok. Marking resolved.